### PR TITLE
Implement undo and redo when removing attachments by code.

### PR DESCRIPTION
**Pull request by SergioEstevao**
This PR implements the capability of undo redo when removing an attachment by pressing on it and select the option remove.

To test:
 - Start the demo app
 - open the demo with content
 - scroll to the coyote image
 - Tap on it
 - Tap again
 - Select Remove Media
 - Check if image was removed
 - Tap Undo 
 - Check if image comes back again.

### CODE REVIEW

File: Aztec/Classes/TextKit/TextView.swift [hunk 1]
```diff
@@ -972,18 +972,28 @@ open class TextView: UITextView {
...
+    /// Removes the attachment that match the attachment identifier provided from the storage
     ///
     /// - Parameter attachmentID: the unique id of the attachment
     ///
     open func remove(attachmentID: String) {
-        storage.remove(attachmentID: attachmentID)
+        guard let range = storage.rangeFor(attachmentID: attachmentID) else {
+            return
+        }
+        let originalText = storage.attributedSubstring(from: range)
```
Comment (diegoreymendez) [inline Aztec/Classes/TextKit/TextView.swift]: From line 983 to line 990 we're repeating code that's in several other places in `TextView.swift`

Why not implement in `TextView.swift` method

```
func replaceCharacters(in: range, with attrString: NSAttributedString)
```

 to take care of unified undo support?

File: Aztec/Classes/TextKit/TextView.swift [hunk 2]
```diff
@@ -972,18 +972,28 @@ open class TextView: UITextView {
         return storage.attachment(withId: id)
     }
 
-    /// Removes the attachments that match the attachament identifier provided from the storage
+    /// Removes the attachment that match the attachment identifier provided from the storage
     ///
     /// - Parameter attachmentID: the unique id of the attachment
     ///
     open func remove(attachmentID: String) {
-        storage.remove(attachmentID: attachmentID)
```
Comment (diegoreymendez) [inline Aztec/Classes/TextKit/TextView.swift]: This method is no longer called from anywhere in the code.  We should remove it from `TextStorage.swift`.

File: Aztec/Classes/TextKit/TextView.swift (hunk 1 above)
Comment (SergioEstevao) [inline Aztec/Classes/TextKit/TextView.swift]: Good suggestion may I do this in another PR just dedicated to that refactor?

Comment (diegoreymendez) [inline Aztec/Classes/TextKit/TextView.swift]: Sure.
